pypeit/core/extract.py

```diff
@@ -175,10 +175,16 @@ def extract_optimal(sciimg, ivar, mask, waveimg, skyimg, thismask, oprof,
                                    np.nansum(thismask_sub[oprof_good,:]*oprof_sub[oprof_good,:]**2, axis=1)
         oprof_bad = badwvs & ((oprof_smash <= 0.0) | (np.isfinite(oprof_smash) == False) | (wave_opt <= 0.0) | (np.isfinite(wave_opt) == False))
         if oprof_bad.any():
+            # If there are no good profile wavelengths, use boxcar wavelengths for these pixels
+            # get boxcar_radius
+            box_radius = spec.BOX_RADIUS
+            box_denom_no_mask = moment1d(waveimg > 0.0, spec.TRACE_SPAT, 2 * box_radius, row=spec.trace_spec)[0]
+            wave_no_mask = moment1d(waveimg, spec.TRACE_SPAT, 2 * box_radius, row=spec.trace_spec)[0] / (
+                        box_denom_no_mask + (box_denom_no_mask == 0.0))
+            wave_opt[oprof_bad] = wave_no_mask[oprof_bad]
             # For pixels with completely bad profile values, interpolate from trace.
-            f_wave = scipy.interpolate.RectBivariateSpline(spec_vec,spat_vec, waveimg*thismask)
-            wave_opt[oprof_bad] = f_wave(spec.trace_spec[oprof_bad], spec.TRACE_SPAT[oprof_bad],
-                                         grid=False)
+            #f_wave = scipy.interpolate.RectBivariateSpline(spec_vec,spat_vec, waveimg*thismask)
```

Comment:
Remove?

Reply:
Let's leave the old code there for now in case we need to revert.